### ISSUE TITLE
Flickr30k Entities fine-tuning clarification

### ISSUE THREAD

**Issue by tonyduan** [uclanlp/visualbert#9]
Hi,

Thanks for the open-source repository.

I was wondering: how did you implement fine-tuning for the Flickr30k Entities dataset? From the ACL short paper:

![Screen Shot 2020-06-01 at 3 25 11 PM](https://user-images.githubusercontent.com/1163598/83460789-16655280-a41c-11ea-85ab-e55285fd2030.png)

What is the loss between the predicted alignment and the ground-truth alignment? As noted in your preprint on arXiv, this can be a bit complicated because the ground-truth alignment can have multiple boxes.

**Comment by liunian-harold-li** [comment]
Hi, thank you for your interest in our work. The loss is implemented at https://github.com/uclanlp/visualbert/blob/master/pytorch_pretrained_bert/modeling.py#L1568. For each entity, we minimize the KL-Divergence between predicted distribution over all detected boxes and the target distribution. For instance, if Entity A can be aligned to BOX1 and BOX2, and we have 36 boxes in total, the target distribution is a categorical distribution over 36 boxes, with BOX1 and BOX2 having probability 0.5 respectively. The part where we prepare the target distribution is implemented at https://github.com/uclanlp/visualbert/blob/master/dataloaders/flickr_dataset.py#L238.

**Comment by tonyduan** [comment]
This is fantastic, thanks for clarifying. Closing this issue.
